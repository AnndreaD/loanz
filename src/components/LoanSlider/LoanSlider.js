import React from 'react';
import PropTypes from 'prop-types';
import { TextField } from '@material-ui/core';
import { Slider, Typography } from '@material-ui/core';
import './LoanSlider.scss';

const LoanSlider = ({ title, value, onChange, min, max, step, marks }) => {
    const handleSliderChange = (event, newValue) => {
        onChange(newValue);
    };

    const handleInputChange = (event) => {
        onChange(event.target.value);
    };

    return (
        <div className="slider-container">
            <Typography id="discrete-slider-small-steps">{title}</Typography>
            <TextField value={value} onChange={handleInputChange} variant="outlined" id="outlined-basic" />
            <div className="loan-slider">
                <Slider
                    defaultValue={value}
                    value={typeof value === 'number' ? value : 0}
                    aria-labelledby="discrete-slider-small-steps"
                    step={step}
                    onChange={handleSliderChange}
                    min={min}
                    max={max}
                    valueLabelDisplay="auto"
                    marks={marks}
                />
            </div>
        </div>
    );
};

export default LoanSlider;

LoanSlider.propTypes = {
    title: PropTypes.string,
    value: PropTypes.number,
    min: PropTypes.number,
    max: PropTypes.number,
    step: PropTypes.number,
    marks: PropTypes.array,
};
