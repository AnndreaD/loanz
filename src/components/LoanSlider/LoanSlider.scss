@import '../../colors';

.slider-container {
    width: 50%;

    .MuiFormControl-root {
        width: 100%;
    }

    .MuiOutlinedInput-root.Mui-focused .MuiOutlinedInput-notchedOutline {
        border-color: $green-profile;
    }
    .fieldset {
        border-color: red;
    }

    .loan-slider {
        width: 100%;
        .MuiSlider-root {
            color: $green-profile;
        }
    }
}
