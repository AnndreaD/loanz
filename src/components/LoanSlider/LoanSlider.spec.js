import LoanSlider from './LoanSlider';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Typography } from '@material-ui/core';
import expect from 'expect';
import '@testing-library/jest-dom';

configure({ adapter: new Adapter() });

const marks = [
    {
        value: 0,
        label: 'Minimum',
    },
    {
        value: 20,
        label: 'Maximum',
    },
];

const defaultTestProps = {
    title: 'This is a title',
    value: 10,
    min: 0,
    max: 20,
    step: 1,
    marks: marks,
};

describe('LoanSlider', () => {
    it('SNAPSHOT: should render expected content', () => {
        const defaultTree = renderer.create(<LoanSlider />);
        expect(defaultTree.toJSON()).toMatchSnapshot();
    });

    it('should render right content', () => {
        const wrapper = mount(<LoanSlider {...defaultTestProps} />);
        const title = wrapper.find(Typography);
        expect(title).toHaveLength(1);
        expect(title.text()).toEqual('This is a title');
    });
});
