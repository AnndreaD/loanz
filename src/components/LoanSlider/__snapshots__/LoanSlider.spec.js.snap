// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoanSlider SNAPSHOT: should render expected content 1`] = `
<div
  className="slider-container"
>
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
    id="discrete-slider-small-steps"
  />
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        id="outlined-basic"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-2"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    className="loan-slider"
  >
    <span
      className="MuiSlider-root MuiSlider-colorPrimary"
      onMouseDown={[Function]}
    >
      <span
        className="MuiSlider-rail"
      />
      <span
        className="MuiSlider-track"
        style={
          Object {
            "left": "0%",
            "width": "0%",
          }
        }
      />
      <input
        type="hidden"
        value="0"
      />
      <span
        aria-labelledby="discrete-slider-small-steps"
        aria-orientation="horizontal"
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-5"
        data-index={0}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        role="slider"
        style={
          Object {
            "left": "0%",
          }
        }
        tabIndex={0}
      >
        <span
          className="PrivateValueLabel-offset-7 MuiSlider-valueLabel"
        >
          <span
            className="PrivateValueLabel-circle-8"
          >
            <span
              className="PrivateValueLabel-label-9"
            >
              0
            </span>
          </span>
        </span>
      </span>
    </span>
  </div>
</div>
`;
